### PR TITLE
[Merged by Bors] - feat(CategoryTheory): pull colimits out of hom functors with Yoneda on the LHS

### PR DESCRIPTION
**Pull request by TwoFX**
A technical result on the way towards showing that Ind-objects are closed under filtered colimits.

---

Australians would probably call this result "the Yoneda lemma".

- [x] depends on: #17407
- [x] depends on: #17438
- [x] depends on: #17439

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Limits/Preserves/Yoneda.lean [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  _ ≅ F.flip ⋙ (whiskeringRight _ _ _).obj uliftFunctor.{u} ⋙ colim :=
+        isoWhiskerLeft F.flip (preservesColimitNatIso uliftFunctor.{u})
+  _ ≅ (yoneda.op ⋙ coyoneda ⋙ (whiskeringLeft _ _ _).obj F) ⋙ colim := isoWhiskerRight
+        (isoWhiskerRight largeCurriedYonedaLemma.symm ((whiskeringLeft _ _ _).obj F)) colim
+  _ ≅ yoneda.op ⋙ colimit (F ⋙ yoneda) :=
+        isoWhiskerLeft yoneda.op (colimitIsoFlipCompColim (F ⋙ yoneda)).symm
+
+theorem yonedaYonedaColimit_app_inv {X : C} : ((yonedaYonedaColimit F).app (op X)).inv =
+    (colimitObjIsoColimitCompEvaluation _ _).hom
+      ≫ (colimit.post F (coyoneda.obj (op (yoneda.obj X)))) := by
```
Comment (joelriou) [inline Mathlib/CategoryTheory/Limits/Preserves/Yoneda.lean:56]:
```suggestion
    (colimitObjIsoColimitCompEvaluation _ _).hom ≫
      (colimit.post F (coyoneda.obj (op (yoneda.obj X)))) := by
```

File: Mathlib/CategoryTheory/Limits/Preserves/Yoneda.lean [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+universe v v₁ v₂ v₃ u u₁ u₂ u₃
+
+namespace CategoryTheory
+
+open CategoryTheory.Limits Opposite
+
+variable {C : Type u} [Category.{v} C]
+
+variable {J : Type v} [SmallCategory J] (F : J ⥤ Cᵒᵖ ⥤ Type v)
```
Comment (joelriou) [inline Mathlib/CategoryTheory/Limits/Preserves/Yoneda.lean:38]: Is there a way to relax the universe assumptions on `J` by requesting that certain limits/colimits of shape `J` exist?

Comment (TwoFX) [reply Mathlib/CategoryTheory/Limits/Preserves/Yoneda.lean:38]: I made an attempt at this, what do you think?

**Comment by joelriou** [comment]
Thanks!

bors merge

**Pull request closed without merging**